solutions/cf-862E.mdx

```diff
@@ -0,0 +1,138 @@
+---
+id: cf-862E
+source: CF
+title: Mahmoud and Ehab and the function
+author: Ryan Chou
+---
+ 
+[Official Analysis (C++)](https://codeforces.com/blog/entry/54590)
+
+## Explanation
+
+Let's expand the rule for $f(j)$.
+
+$f(j) = |(\sum_{i = 1}^{n} (-1)^{i - 1} \cdot a_i) - (\sum_{i = 1}^{n} (-1)^{i - 1} \cdot b_{i + j})|$
+
+Let's call the added value $k$ and the $a$'s alternating sum $v$.
+
+Note that for each query, adding $k$ to an even-length segment won't make a difference because they'll cancel themselves. Otherwise, $v$ will differ by $k$ (it's not canceled by the following element).
+
+This means that we can precompute $v$ in $\mathcal{O}(N)$ time.
+
+What about $b$? We can compute each segment of $n$ elements with two pointers, since $b$'s values never change.
+
+However, checking every value of $j$ will surely TLE. Instead, we can notice that since we're calculating absolute value, we only care about the two closest elements on either side of it. Thus, we'll binary search on the alternating sums of $b$.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N + (M + Q)\log M)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+#include <algorithm>
+using std::vector;
+
+int main() {
+	int n, m, q;
+	std::cin >> n >> m >> q;
+	
+	vector<int> a(n), b(m);
+
+	for (int i = 0; i < n; i++) {
+		std::cin >> a[i];
+	}
+	for (int i = 0; i < m; i++) {
+		std::cin >> b[i];
+	}
+
+	/*
+	 * calculate the odd and even sums, 
+	 * we care about their sums in the case 
+	 * where we start on a index with a different parity
```

Comment:
i'd like if you could be a tad bit more formal in your comments
i'm fine with this but it just contradicts with the mood of the editorial